### PR TITLE
Fast multiple setEx with the pipeline transcation (RedisAdapter::doSave)

### PR DESCRIPTION
**Pull request by masterklavi**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | no |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/Cache/Adapter/RedisAdapter.php [hunk 1]
```diff
@@ -93,22 +93,24 @@ protected function doSave(array $values, $lifetime)
...
 
-        if (!$this->redis->mSet($serialized)) {
-            return false;
-        }
-
-        if ($lifetime >= 1) {
-            foreach ($serialized as $id => $v) {
-                $this->redis->expire($id, $lifetime);
+        if ($lifetime > 0) {
+            $pipe = $this->redis->multi(Redis::PIPELINE);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Adapter/RedisAdapter.php]: missing `\`: `\Redis`

Comment (masterklavi) [inline src/Symfony/Component/Cache/Adapter/RedisAdapter.php]: Ok, fixed.

**Comment by staabm** [comment]
did you profile how much faster it is?

**Comment by masterklavi** [comment]
Yes. The new method is faster in ~4 times.

**Comment by theofidry** [comment]
Is it possible to have a Backfire profile of it?

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by masterklavi** [comment]
> Is it possible to have a Backfire profile of it?

https://blackfire.io/profiles/6a1e8027-1375-4a93-b46e-14ccbc2c639a/graph

**Comment by stof** [comment]
@masterklavi what we would need is a backfire comparison between profiles of the old code and the new code. This would be much easier to analyze than some code running both methods one after the other with different names.

**Comment by masterklavi** [comment]
Ok. https://blackfire.io/profiles/compare/b801ec2e-967b-4961-9b28-f96db157a212/graph

**Comment by nicolas-grekas** [comment]
Thank you @masterklavi.

**Pull request closed without merging**